internal/builders/generic/README.md

```diff
@@ -226,6 +226,7 @@ The [generic workflow](https://github.com/slsa-framework/slsa-github-generator/b
 | -------------------- | -------- | ----------------------------------------------------------------------------------------------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
 | `base64-subjects`    | yes      |                                                                                                 | Artifact(s) for which to generate provenance, formatted the same as the output of sha256sum (SHA256 NAME\n[...]) and base64 encoded. The encoded value should decode to, for example: `90f3f7d6c862883ab9d856563a81ea6466eb1123b55bff11198b4ed0030cac86 foo.zip` |
 | `upload-assets`      | no       | false                                                                                           | If true provenance is uploaded to a GitHub release for new tags.                                                                                                                                                                                                 |
+| `upload-tag-name`    | no       |                                         | Set when `upload-assets` is true and workflow is **not** triggered by a release/tag. Can also be used to override the default release name on tagged events.
```

Comment:
If specified, the `upload-tag-name` is used if `upload-assets` is true regardless of the trigger.

You might want to look at the doc for the input on the workflow itself.
https://github.com/slsa-framework/slsa-github-generator/blob/3dd7043831da4c2b87e7be09a3dfe06806cbb81e/.github/workflows/generator_generic_slsa3.yml#L46-L49